File: VB/E5101/Controllers/PdfViewerController.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc
Imports E5101.Models
Imports DevExpress.Pdf
Imports System.IO

Namespace E5101.Controllers

    Public Class PdfViewerController
        Inherits System.Web.Mvc.Controller

        Public Const SESSION_KEY As String = "PdfFile"

        Public Function PreviewByFileName(ByVal pdfFileName As String) As ActionResult
            Using stream As System.IO.FileStream = New System.IO.FileStream(Me.Server.MapPath(pdfFileName), System.IO.FileMode.Open)
                Dim pdfFileBytes As Byte() = New Byte(stream.Length - 1) {}
                stream.Read(pdfFileBytes, 0, CInt(stream.Length))
                Me.Session(E5101.Controllers.PdfViewerController.SESSION_KEY) = pdfFileBytes
            End Using

            Return Me.PartialView("_PdfViewerPartial")
        End Function

        Public Function Preview() As ActionResult
            Return Me.PartialView("_PdfViewerPartial")
        End Function

        Public Function DocumentViewPartial() As ActionResult
            Dim documentProcessor As DevExpress.Pdf.PdfDocumentProcessor = New DevExpress.Pdf.PdfDocumentProcessor()
            Using stream As System.IO.MemoryStream = New System.IO.MemoryStream(CType(Me.Session(E5101.Controllers.PdfViewerController.SESSION_KEY), Byte()))
                documentProcessor.LoadDocument(stream)
            End Using

            Dim model As System.Collections.Generic.List(Of E5101.Models.PdfPageModel) = New System.Collections.Generic.List(Of E5101.Models.PdfPageModel)()
            For pageNumber As Integer = 1 To documentProcessor.Document.Pages.Count
                model.Add(New E5101.Models.PdfPageModel(documentProcessor) With {.PageNumber = pageNumber})
            Next

            Return Me.PartialView("_DocumentViewPartial", model)
        End Function
    End Class
End Namespace

File: CS/E5101/Controllers/PdfViewerController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using DevExpress.Web.Mvc;
using E5101.Models;
using DevExpress.Pdf;
using System.IO;

namespace E5101.Controllers {
    public class PdfViewerController : Controller {
        public const string SESSION_KEY = "PdfFile";

        public ActionResult PreviewByFileName(string pdfFileName) {
            Session[SESSION_KEY] = System.IO.File.ReadAllBytes(Server.MapPath(pdfFileName));
            return PartialView("_PdfViewerPartial");
        }

        public ActionResult Preview() {
            return PartialView("_PdfViewerPartial");
        }

        public ActionResult DocumentViewPartial() {
            PdfDocumentProcessor documentProcessor = new PdfDocumentProcessor();

            MemoryStream stream = new MemoryStream((byte[])Session[SESSION_KEY]);
            documentProcessor.LoadDocument(stream);

            List<PdfPageModel> model = new List<PdfPageModel>();
            for (int pageNumber = 1; pageNumber <= documentProcessor.Document.Pages.Count; pageNumber++) {
                model.Add(new PdfPageModel(documentProcessor) {
                    PageNumber = pageNumber
                });
            }
            return PartialView("_DocumentViewPartial", model);
        }
    }
}
